src/git_provider/github.rs

```diff
@@ -12,6 +23,59 @@ impl GitHub {
 }
 
 impl GitProvider for GitHub {
+    fn set_token(&self, path: &PathBuf) -> Result<String> {
+        let token = dialoguer::Input::<String>::with_theme(&ColorfulTheme::default())
+            .with_prompt(
+                "Set up your GitHub personal access token. You can generate one at:
+https://github.com/settings/tokens/new?description=Gi&scopes=repo,read:org,read:user,user:email\n",
+            )
+            .interact()?;
+
+        let data = json!({
+            "accessToken": token
+        });
+
+        fs::write(path, data.to_string())
+            .context("Failed to write to token file")
+            .suggestion("Check if you have write permissions")?;
+
+        Ok(token)
+    }
+
+    fn get_token(&self) -> Result<String> {
+        let home_dir = dirs::home_dir().ok_or_eyre("Failed to get home directory")?;
+        let token_file = home_dir.join(".config/gi/token");
```

Comment:
I think this won't work on windows because you are assuming `/` as a separator. A better solution would look like:

```suggestion
        let token_file = home_dir.join(".config").push("gi").push("token");
```

I'm not sure if it compiles as I have not tested this.

Reply:
You're perfectly right. I think I was too tired to think about that😅. I'll change it before merging